### DISCUSSION TITLE
Containers not starting due to issue with "extra_hosts"

### DISCUSSION THREAD

**Discussion by sjdwhiting** [Security-Onion-Solutions/securityonion#3621] [Unsupported Versions]
Hello all,

I am attempting to set up a new distributed install. I've done several installs of SO at this point but I've run into a new issue I can't seem to resolve.

I recently attempted to add a search node to my setup which was previously working with a manager and heavy node. After some failed attempts and other issues, I seemingly got it working but now it is getting hung up while applying the elasticsearch salt state. I went to my manager node and found that after a reboot, so-elasticsearch was also stuck at the same point. so-soc is also hanging. 

I attempted so-soc-restart and received the following error. 

`          ID: so-soc
    Function: docker_container.running
      Result: False
     Comment: Failed to translate input. Additional info follows:

              invalid:
                  ----------
                  extra_hosts:
                      '4' is not a valid IP address

              If you feel this information is incorrect, the skip_translate argument can be used to skip input translation for the argument(s) identified as invalid. See the documentation for details.
     Started: 18:21:13.738761
    Duration: 13.983 ms
     Changes:
----------`

When I run so-elasticsearch-restart, I get the following at the tail end. Also, the last line is where it is hanging. That line also appears at the end of sosetup.log on the new search node.

```
`[INFO    ] Running state [so-elasticsearch] at time 18:35:09.451310
[INFO    ] Executing state docker_container.running for [so-elasticsearch]
[ERROR   ] Failed to translate input. Additional info follows:

invalid:
    ----------
    extra_hosts:
        '4' is not a valid IP address

If you feel this information is incorrect, the skip_translate argument can be used to skip input translation for the argument(s) identified as invalid. See the documentation for details.
[INFO    ] Completed state [so-elasticsearch] at time 18:35:09.466262 (duration_in_ms=14.952)
[INFO    ] Running state [/opt/so/conf/so-status/so-status.conf] at time 18:35:09.466526
[INFO    ] Executing state file.append for [/opt/so/conf/so-status/so-status.conf]
[INFO    ] File /opt/so/conf/so-status/so-status.conf is in correct state
[INFO    ] Completed state [/opt/so/conf/so-status/so-status.conf] at time 18:35:09.475276 (duration_in_ms=8.749)
[INFO    ] Running state [/usr/sbin/so-elasticsearch-templates-load] at time 18:35:09.476207
[INFO    ] Executing state cmd.run for [/usr/sbin/so-elasticsearch-templates-load]
[INFO    ] Executing command '/usr/sbin/so-elasticsearch-templates-load' in directory '/opt/so'`
```

I am guessing that this '4' appeared because of the failed install attempts but I'm not sure where to go to remove it. At this point my install is not working at all. 

Any assistance is much appreciated,

Thanks,

Sebastian

**Comment by wtl06639** [comment]
I just opened a ticket with the same issue but with 192. I am not sure where it is getting the 192 from or where to go to delete it. Any help would be appreciated .

**Comment by defensivedepth** [comment]
What version are you on? ISO or Network Install (Ubuntu/CentOS?)

**Reply by wtl06639** [reply]
Using the ISO - 2.3.50 .  I did a bunch of troubleshooting last evening. I'm not quite sure why the initial setup isn't working properly. But I managed to manually fix it, though i wont know if i got everything. I have a couple more sensors i have to do so i will see if they have issues too or not. The first one had no problems, the second one did.

Heres what i had to do to fix and get it start showing up on the grid and running everything properly and have the manager so-services start working properly again (because of the botched install/registration with manager, elastalert, elasticsearch, and soc containers  failed to to start on manager)

On the manager: 
cd to opt/so/saltstack/local/pillar/data/nodestab.sls and sensorstab.sls and correct the IP of the Node. Mine some how was added just as 192. 

Also go to /opt/so/saltstack/local/pillar/minions/<sensorname.sls> and update the blank spots where the IP should be based off a what a working node shows. All these appeared to be empty for me

run so-docker-refresh

manager so-services should start working again,

then on the sensor go to /home/<username>/SecurityOnion/ and run so-setup-network and the sensor tried to run the setup again, Appeared to work better this time. I did run into some hiccups of all the docker containers not appears to start on their on. We will see if that continues. But I was at least able to manually start them successfully. And it it now showing up in the grid and showing alerts from this sensor.

**Reply by defensivedepth** [reply]
Please run the following on your manager:

`sudo salt-call grains.get ip_interfaces`

**Comment by wtl06639** [comment]
This issue has come back for me again, but now saying:

         ID: so-elasticsearch
    Function: docker_container.running
      Result: False
     Comment: Failed to translate input. Additional info follows:

              invalid:
                  ----------
                  extra_hosts:
                      'OrderedDict([('nsmanager', None)])' is not a key:value pair

              If you feel this information is incorrect, the skip_translate argument can be used to skip input translation for the argument(s) identified as invalid. See the documentation for details.
     Started: 11:53:02.929053
    Duration: 12.993 ms
     Changes:

So not sure where this problem is coming from.  I run sudo salt-call grains.get ip_interfaces and it returned the correct IPs. All appears correct under the manager file under /opt/so/saltstack/local/pillar/minions directory, and under the /opt/so/saltstack/local/pillar/data
directory. So not sure what's causing it now.

**Reply by defensivedepth** [reply]
Please post the (sanitized if needed) output of `sudo salt-call grains.get ip_interfaces`

**Comment by wtl06639** [comment]
Here is the output (I changed the ens192 IP from the correct one) : 

local:
    ----------
    docker0:
        - 172.17.0.1
    ens192:
        - 192.168.1.2
    lo:
        - 127.0.0.1
    veth288a37c:
    veth2958fdd:
    veth48a64ec:
    veth499fa7f:
    veth4cc1193:
    veth5f5b841:
    veth682426f:
    veth704498f:
    veth8e150f6:
    veth9f97912:
    vethaddcf98:
    vethb0dee6e:
    vethc043c1c:
    vethc2f78b0:
    vethd04d3ba:
    vethe282bca:
    vethe2b716f:
    vethf532969:
    vethf54b8b7:

**Reply by defensivedepth** [reply]
Are you using static IPs or DHCP?

**Reply by wtl06639** [reply]
I am using static IPs

**Reply by m0duspwnens** [reply]
@wtl06639
Can you check your files under `/opt/so/saltstack/local/pillar/data/managertab.sls` and verify that the proper IP info and name is in there?
